activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -600,6 +601,13 @@ def initialize_type_map(m = type_map)
 
           m.register_type %r(^enum)i, Type.lookup(:string, adapter: :mysql2)
           m.register_type %r(^set)i,  Type.lookup(:string, adapter: :mysql2)
+          m.register_type(%r(^json)i) do |sql_type|
+            if /^json/.match?(sql_type)
+              MySQL::Json.new
```

Comment:
Referencing the `MySQL` constant directly here seems like an anti-pattern to me, but I could definitely be wrong. Just by looking at the rest of this adapter anyways.

There is an explicit inclusion of [`MySQL::Quoting`](https://github.com/rails/rails/blob/e1cd663051b3b21511f7f5986cdca232aba6c136/activerecord/lib/active_record/connection_adapters/mysql/quoting.rb) and [`MySQL::SchemaStatements`](https://github.com/rails/rails/blob/e1cd663051b3b21511f7f5986cdca232aba6c136/activerecord/lib/active_record/connection_adapters/mysql/schema_statements.rb) though so maybe it's ok? I would have to look at the other adapters to feel more confident about this change.

Reply:
@zzak thanks for notice, but I thought about this: look at the `MySQL::SchemaStatements` file by the link you sent: there are `MySQL::TypeMetadata`, `MySQL::SchemaCreation `, `MySQL::SchemaDumper `, `MySQL::TableDefinition ` and  `MySQL::Column` constants (this choice is mostly influenced by it)

Reply:
@razum2um Ok, but aren't those all constants from within Rails? Where does the `MySQL::Json` constant come from? 🤔 Sorry if this question is off track, I'm just trying to make sure I understand what is going on here.

Reply:
@zzak `MySQL::Json` is a new class in this PR :)

Reply:
🤦